src/lib/learnerrecord.ts

```diff
@@ -357,6 +361,21 @@ export class CourseRecord {
 		}
 		return startedDate
 	}
+
+	async getAvailableCompletedMandatoryModulesForCourse(user: model.User) {
+		const availableModuleIds = await this.getAvailableModuleIdsForCourse(user)
+
+		return this.getCompletedModules()
+			.filter(module => module.optional === false)
+			.filter(module => availableModuleIds.includes(module.moduleId))
+	}
+
+	async getAvailableModuleIdsForCourse(user: model.User) {
+		const courseFromCatalog = await catalog.get(this.courseId, user)
```

Comment:
Could we use on the course data from the top-level `home.ts home` function for this? Seems like this will introduce a `GET /course/{courseID}` catalogue API call for each course processed?

Reply:
Good point! 👀